app/models/course.rb

```diff
@@ -282,6 +284,17 @@ def tracked_revisions
              .where(wiki_id: wiki_ids)
   end
 
+  def tracked_namespaces
```

Comment:
I don't like this data structure very much. At least on the Ruby side of things, I think it will be easier to work with the `courses_namespaces` objects directly.

For example, in `UpdateNamespaceStats#update_stats`, you don't need to have nested loops if you do something like:
```
@course.courses_namespaces.each do |c_ns|
  wiki = c_ns.wiki
  namespace = c_ns.namespace
  ...
end

Similarly, you can probably serialize the courses_namespaces objects more directly, instead of doing this somewhat complex mapping.